sound/soc/codecs/rt715.c

```diff
@@ -319,6 +319,8 @@ static int rt715_mux_get(struct snd_kcontrol *kcontrol,
 	/* FIXME: PLB: check return status on read/write */
 	snd_soc_component_write(component, reg, 0x0);
 	snd_soc_component_read(component, RT715_READ_HDA_0, &val);
+	if ((e->reg == 0x24 || e->reg == 0x25) && (val > 0))
+		val -= 1;
```

Comment:
with all due respect, the code here doesn't look so good. can we try to improve it a bit for upstream?

Reply:
@plbossart  Could you help review it to see if this format is suitable for upstream? if not, could you show some example code for reference so I can do modification according to it, thanks.